Added morpho.plot.histo2D_divergence method to create a 2D plot where...

joejohnston203 opened the pull request:
points with divergence==1 are in a different color

Updated morpho_linear_fit.yaml to include this 2D divergence plot
(although the default example has not points with divergence==1).

The user can set the color for divergence==0 and divergence==1 plots.

Also fixed a bug in _get2DHisto() where the y_range and x_range
were equal in some cases.

joejohnston203 commented:
[histo_divergence_a_b_alb=5.pdf](https://github.com/project8/morpho/files/1233361/histo_divergence_a_b_alb.5.pdf)

This plot was made by forcing "a" to be greater than 5 in the stan model.

guiguem commented on morpho/plot/histo.py:
```diff
@@ -339,9 +370,84 @@ def histo2D(param_dict):
+    if(list_dataX_div0 is None):
+        return
+
+    xmin = min(list_dataX_div0+list_dataX_div1)
```
shouldn't this be xmin = min(list_dataX_div0, list_dataX_div1) instead of xmin = min(list_dataX_div0+list_dataX_div1)? ( comma instead of +)
same below?

guiguem commented on morpho/plot/histo.py:
```diff
@@ -15,6 +15,7 @@
 - histo: plot a 1D histogram using a list of data (x)
 - spectra: plot a 1D histogram using two lists of data (x,bin_content)
 - histo2D: plot a 2D histogram using two lists of data (x,y)
+- histo2D_divergence
```
let's add a light description

joejohnston203 commented:
I added a comment describing histo2D_divergence.

If list_dataX_div0, etc, are python lists, then adding them concatenates the elements and min(list_div0+list_div1) gives the smallest element as expected. But if they were switched to numpy arrays, this would break. min(list_div0, list_div1) returns the list with the smallest element rather than returning the smallest element itself. I have updated the code to use min(min(list_div0), min(list_div1)) so that it will work with python lists or numpy arrays.
